sherpa/astro/optical/__init__.py

```diff
@@ -56,10 +56,10 @@
 
 __all__ = ('AbsorptionEdge', 'AccretionDisk', 'AbsorptionGaussian',
            'AbsorptionLorentz', 'EmissionLorentz', 'OpticalGaussian',
-           'EmissionGaussian', 'AbsorptionVoigt', 'BlackBody',
+           'EmissionGaussian', 'BlackBody',
            'Bremsstrahlung', 'BrokenPowerlaw', 'CCM', 'LogAbsorption',
            'LogEmission', 'Polynomial', 'Powerlaw', 'Recombination',
-           'EmissionVoigt', 'XGal', 'FM', 'LMC', 'SM', 'SMC', 'Seaton')
+           'XGal', 'FM', 'LMC', 'SM', 'SMC', 'Seaton')
```

Comment:
We want to leave Absorption/EmissionVoigt available so that users who have used these models get some information on what has changed. I suggest

```
diff --git a/sherpa/astro/optical/__init__.py b/sherpa/astro/optical/__init__.py
index 15112f88..42807644 100644
--- a/sherpa/astro/optical/__init__.py
+++ b/sherpa/astro/optical/__init__.py
@@ -41,6 +41,7 @@ import numpy
 from sherpa.models.parameter import Parameter, tinyval
 from sherpa.models.model import ArithmeticModel, RegriddableModel1D
 from sherpa.utils import SherpaFloat, sao_fcmp
+from sherpa.utils.err import ModelErr

 _tol = numpy.finfo(numpy.float32).eps

@@ -57,6 +58,7 @@ _tol = numpy.finfo(numpy.float32).eps
 __all__ = ('AbsorptionEdge', 'AccretionDisk', 'AbsorptionGaussian',
            'AbsorptionLorentz', 'EmissionLorentz', 'OpticalGaussian',
            'EmissionGaussian', 'BlackBody',
+           'AbsorptionVoigt', 'EmissionVoigt',  # Use Voigt1D instead
            'Bremsstrahlung', 'BrokenPowerlaw', 'CCM', 'LogAbsorption',
            'LogEmission', 'Polynomial', 'Powerlaw', 'Recombination',
            'XGal', 'FM', 'LMC', 'SM', 'SMC', 'Seaton')
@@ -96,6 +98,19 @@ def _extinct_interp(xtable, etable, x):
     return out


+class AbsorptionVoigt(ArithmeticModel):
+    """This model has been replaced by Voigt1D"""
+
+    def __init__(self, *args):
+        raise ModelErr("The AbsorptionVoigt model has been replaced by Voigt1D")
+
+class EmissionVoigt(ArithmeticModel):
+    """This model has been replaced by Voigt1D"""
+
+    def __init__(self, *args):
+        raise ModelErr("The EmissionVoigt model has been replaced by Voigt1D")
+
+
 # This model sets in edge (in Angstroms) beyond which absorption
 # is a significant feature to the spectrum or SED.
 class AbsorptionEdge(RegriddableModel1D):
```

Reply:
Haven't tried it yet but I suspect the test of the file ```sherpa/astro/optical/tests/test_optical.py``` would fail

Reply:
========================================== FAILURES ==========================================
____________________________ test_models.test_create_and_evaluate ____________________________

self = <test_optical.test_models testMethod=test_create_and_evaluate>

    def test_create_and_evaluate(self):
        x = arange(4500.0,5500.0,1,SherpaFloat)
        count = 0

        for cls in dir(models):
            clsobj = getattr(models, cls)

            if not isinstance(clsobj, type) \
                    or not issubclass(clsobj, ArithmeticModel) \
                    or clsobj in self.excluded_model_classes:
                continue

>           m = clsobj()

sherpa/astro/optical/tests/test_optical.py:44:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError("'AbsorptionVoigt' object has no attribute 'name'") raised in repr()]
AbsorptionVoigt object at 0x7fc3a2d06250>
args = ()

    def __init__(self, *args):
>       raise ModelErr("The AbsorptionVoigt model has been replaced by Voigt1D")
E       sherpa.utils.err.ModelErr: The AbsorptionVoigt model has been replaced by Voigt1D

sherpa/astro/optical/__init__.py:105: ModelErr
================================ 1 failed, 25 passed in 0.12s ================================


But we can adjust the test to handle these two models in the short term

Reply:
It's as simple as

```
% git diff sherpa/astro/optical/tests/test_optical.py
diff --git a/sherpa/astro/optical/tests/test_optical.py b/sherpa/astro/optical/tests/test_optical.py
index 4548de45..ddfb444c 100644
--- a/sherpa/astro/optical/tests/test_optical.py
+++ b/sherpa/astro/optical/tests/test_optical.py
@@ -27,7 +27,9 @@ from sherpa.models.model import ArithmeticModel, RegriddableModel1D, Regriddable


 class test_models(SherpaTestCase):
-    excluded_model_classes = (ArithmeticModel, RegriddableModel1D, RegriddableModel2D)
+    # The Absorption/EmissionVoigt models will be removed at some
+    # point after Sherpa 4.13 is released.
+    excluded_model_classes = (ArithmeticModel, RegriddableModel1D, RegriddableModel2D, models.AbsorptionVoigt, models.EmissionVoigt)

     def test_create_and_evaluate(self):
         x = arange(4500.0,5500.0,1,SherpaFloat)
```